src/lib/io/sol-iio.c

```diff
@@ -315,7 +320,53 @@ create_sysfs_trigger(struct sol_iio_device *device)
 }
 
 static bool
-check_trigger(struct sol_iio_device *device)
+create_hrtimer_trigger(struct sol_iio_device *device, const char *trigger_name)
+{
+    int success, id, len, r = false;
+    char *name, *replace;
+    char path[PATH_MAX];
+
+    if (check_file_existence(CONFIGFS_IIO_TRIGGERS_PATH)) {
+        if (0 == strlen(trigger_name)) {
+            id = rand();
+            len = asprintf(&name, "%d", id);
+        } else
+            len = asprintf(&name, "%s", trigger_name);
+
+        if (len < 0)
+            goto error;
+
+        r = craft_filename_path(path, sizeof(path), CONFIGFS_IIO_TRIGGER_PATH, name);
+        if (!r) {
+            free(name);
+            goto error;
+        }
+
+        success = mkdir(path, S_IRWXU);
+        if (success == 0) {
+            r = set_current_trigger(device, name);
+            if (r)
+                device->trigger_name = name;
+            else {
+                free(name);
```

Comment:
Tip: you can avoid all these `free(name)` on error if you initialise `name` to NULL, and always `free` it on label `error`.